### PR TITLE
[Semi-modular] Replaces the AI's transformer with one that is server-policy friendly

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request

Replaces the AI's transformer with one that creates cyborgs from ghosts , instead of humans.

## Why It's Good For The Game

So the AI's may freely place transformers without fear of bans.

## Changelog
:cl:
tweak:  AI Transformer is now replaced with a new one that is server-policy friendly.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	icon = 'icons/obj/recycling.dmi'
+	icon_state = "separator-AO1"
+	layer = ABOVE_ALL_MOB_LAYER // Overhead
+	density = TRUE
+	/// How many cyborgs are we storing
+	var/stored_cyborgs = 1
+	/// How many cyborgs can we store?
+	var/max_stored_cyborgs = 4
+	/// How much between the construction of a cyborg?
+	var/cooldown_duration = 3000 // 5 minutes
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm]:
```suggestion
	var/cooldown_duration = 5 MINUTES
```

File: code/modules/antagonists/traitor/equipment/Malf_Modules.dm [hunk 2]
```diff
@@ -579,7 +579,7 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 		active = FALSE
 		return
 	var/turf/T = get_turf(owner_AI.eyeobj)
-	var/obj/machinery/transformer/conveyor = new(T)
+	var/obj/machinery/transformer_rp/conveyor = new(T) // Skyrat edit , replaced with our own transformer version.
```
Comment (Gandalf2k15) [inline code/modules/antagonists/traitor/equipment/Malf_Modules.dm]:
```suggestion
	var/obj/machinery/transformer_rp/conveyor = new(T) //SKYRAT EDIT CHANGE - SILLICONQOL - ORIGINAL: var/obj/machinery/transformer/conveyor = new(T)
```

File: modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	if(machine_stat & (BROKEN|NOPOWER))
+		icon_state = "separator-AO0"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/transformer_rp/attack_ghost(mob/dead/observer/target_ghost)
+	. = ..()
+	create_a_cyborg(target_ghost)
+
+/obj/machinery/transformer_rp/process()
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm:48]: Perhaps give a ghost prompt once it's ready for a new cyborg?

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm (hunk 3 above)
Comment (MLGTASTICa) [reply modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm:48]: added it now , accidentally deleted it by accident.

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by Gandalf2k15** [comment]
I don't even think this needs a TM, we'll full merge it because having the other version is b a d.

**Pull request merged**